apps/test-suite/tests/NewNotifications.js

```diff
@@ -407,6 +410,92 @@ export async function test(t) {
+        handleSuccessSpy = t.jasmine.createSpy('handleSuccessSpy');
+        handleErrorSpy = t.jasmine.createSpy('handleErrorSpy').and.callFake((...args) => {
+          console.log(args);
+        });
+        notificationReceivedSpy = t.jasmine.createSpy('notificationReceivedSpy');
+        Notifications.setNotificationHandler({
+          handleNotification: handleNotificationSpy,
+          handleSuccess: handleSuccessSpy,
+          handleError: handleErrorSpy,
+        });
+        subscription = Notifications.addNotificationReceivedListener(notificationReceivedSpy);
+      });
+
+      t.afterEach(() => {
+        if (subscription) {
+          subscription.remove();
+          subscription = null;
+        }
+        Notifications.setNotificationHandler(null);
+        handleNotificationSpy = null;
+        handleSuccessSpy = null;
+        handleErrorSpy = null;
+        notificationReceivedSpy = null;
+      });
+
+      t.it(
+        'shows the notification',
+        // without async-await the code is executed immediately after opening the screen
+        async () =>
+          await new Promise((resolve, reject) => {
+            const secondsToTimeout = 5;
+            let notificationSent = false;
+            Alert.alert(`Please move the app to the background and wait for 5 seconds`);
+            let userInteractionTimeout = null;
+            async function handleStateChange(state) {
+              const identifier = 'test-interactive-notification';
+              if (state === 'background' && !notificationSent) {
+                if (userInteractionTimeout) {
+                  clearInterval(userInteractionTimeout);
+                  userInteractionTimeout = null;
+                }
+                await Notifications.scheduleNotificationAsync(
+                  identifier,
+                  {
+                    title: 'Hello from the application!',
+                    message:
+                      'You can now return to the app and let the test know the notification has been shown.',
+                  },
+                  { seconds: 1 }
+                );
+                notificationSent = true;
+              } else if (state === 'active' && notificationSent) {
+                const notificationWasShown = await askUserYesOrNo('Was the notification shown?');
+                t.expect(notificationWasShown).toBeTruthy();
+                t.expect(handleNotificationSpy).not.toHaveBeenCalled();
+                t.expect(handleSuccessSpy).not.toHaveBeenCalled();
+                t.expect(handleErrorSpy).not.toHaveBeenCalledWith(identifier);
+                t.expect(notificationReceivedSpy).not.toHaveBeenCalled();
+                AppState.removeEventListener('change', handleStateChange);
+                resolve();
+              }
+            }
+            userInteractionTimeout = setTimeout(() => {
+              AppState.removeEventListener('change', handleStateChange);
+              Alert.alert(
```

Comment:
might be worth printing an error or log to the console in this case, as well, so that the proviso is recorded in the test results.

Reply:
Fixed!